src/Lean/Meta/ACLt.lean

```diff
@@ -66,7 +66,9 @@ where
       | .none => return e
 
   lt (a b : Expr) : MetaM Bool := do
-    if ptrAddrUnsafe a == ptrAddrUnsafe b then
+    if a == b then
+      -- We used to have an "optimization" using only pointer equality.
+      -- This was a bad idea, `==` is often much cheaper than `acLt`.
```

Comment:
A reader might think that the pointer equality optimization was a bad idea (and not the “only”)

```suggestion
      -- This was insufficient, as `==` is often much cheaper than `acLt`.
```